dsc/src/subcommand.rs

```diff
@@ -667,7 +668,8 @@ fn list_resources(dsc: &mut DscManager, resource_name: Option<&String>, adapter_
                     exit(EXIT_JSON_ERROR);
                 }
             };
-            write_output(&json, format);
+            write_object(&json, format, !first);
```

Comment:
nit: considering swapping the `true` and `false` order of `first` (and maybe renaming) so that `!first` doesn't have to be evaluated each iteration

```
let mut include_separator = false;
write_object(&json, format, include_separator);
include_separator = true;
```